File: 프로그래머스/Lv.1/완주하지 못한 선수.py
def solution(participant, completion):
    
    participant_dic = {}
    for i in participant:
        if i in participant_dic:
            participant_dic[i] += 1
        else:
            participant_dic[i] = 1
            
    completion_dic = {}
    for i in completion:
        if i in completion_dic:
            completion_dic[i] += 1
        else:
            completion_dic[i] = 1
            
    if completion_dic.keys() == participant_dic.keys():
        for i in completion_dic.keys():
            if completion_dic[i] != participant_dic[i]:
                return i
    else:
        for i in participant_dic.keys():
            if i not in completion_dic.keys():
                return i

File: 프로그래머스/Lv.1/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> participantMap = new HashMap<>();
        Map<String, Integer> completionMap = new HashMap<>();
        
        for (String person: participant) {
            if (participantMap.containsKey(person)) {
                participantMap.put(person, participantMap.get(person) + 1);
            } else {
                participantMap.put(person, 1);
            }
        }
        
        for (String person: completion) {
            if (completionMap.containsKey(person)) {
                completionMap.put(person, completionMap.get(person) + 1);
            } else {
                completionMap.put(person, 1);
            }
        }
        
        for (String person: participant) {
            if (!completionMap.containsKey(person) || !completionMap.get(person).equals(participantMap.get(person))) {
                return person;
            }
        }
        return "";
    }
}
